django/contrib/postgres/signals.py

```diff
@@ -13,6 +13,10 @@ def register_hstore_handler(connection, **kwargs):
             register_hstore(connection.connection, globally=True, unicode=True)
         else:
             register_hstore(connection.connection, globally=True)
+        with connection.cursor() as cursor:
+            cursor.execute("""SELECT oid from pg_type WHERE typname = 'hstore'""")
+            oid = cursor.fetchone()[0]
+            connection.introspection.data_types_reverse[oid] = 'django.contrib.postgresql.fields.HStoreField'
```

Comment:
It would be great if this was only executed when required.

Do you think making `DatabaseIntrospection.data_types_reverse` a `cached_property` fetching the correct oid could work?